dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -271,6 +271,19 @@ def replace_hostname(url)
   @browser.execute_script %q-sessionStorage.setItem('progress', '{"hourofcode":{"' + appOptions.serverLevelId + '":100}}')-
 end
 
+Then /^validate progress for level is "([^"]*)"/ do |test_result|
+  steps %{
+    And I reload the page
+    And I wait to see ".header_middle"
+    And I wait for 10 seconds
+    And element ".header_middle a.level_link:last" has class "#{test_result}"
+    Then I am on "http://studio.code.org/s/hourofcode"
+    And I wait to see ".user-stats-block"
+    And I wait for 10 seconds
+    And element ".user-stats-block .games:first a.level_link:last" has class "#{test_result}"
+  }
+end
```

Comment:
I'm concerned that this step definition is misleading on a few levels:

* It's specific to the `hourofcode` script (hard-coded line 280); it would be easy for someone testing a different script to try and use this step, and be surprised when it didn't work.  Maybe put `hourofcode` in the name?
* It always checks the last level of the script (`a.level_link:last` selector) which is not implied by the step name.
* It navigates to a new page and leaves you there, which feels like a side effect given the step name.  Now I understand why you only use this as the very last step in a scenario.
* I'd prefer arbitrary 10-second waits to be replaced with waiting for something specific as much as possible; if not possible here, then a comment explaining what the 10 seconds is for would be helpful.

I'm not sure we need to make this fully generic yet, but if not I feel like the step name should do a little more to call out how specific this is.